xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/html/DefaultHTMLCleaner.java

```diff
@@ -155,7 +155,7 @@ public Document clean(Reader originalHtmlContent, HTMLCleanerConfiguration confi
             cleanedNode.setDocType(new DoctypeToken("html", "PUBLIC", "-//W3C//DTD XHTML 1.0 Strict//EN",
                 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"));
             result =
-                new XWikiDOMSerializer(cleanerProperties).createDOM(getAvailableDocumentBuilder(), cleanedNode);
+                new DomSerializer(cleanerProperties).createDOM(cleanedNode);
```

Comment:
Note that not reusing the context DocumentBuilder will have a performance impact but I guess there is not much we can do about it.

Reply:
>  I guess there is not much we can do about it.

I just checked and indeed the document builder is created inside the method so I don't really see how to provide it.